enterprise/server/cmd/ci_runner/main.go

```diff
@@ -2084,14 +2110,19 @@ func getStructuredCommandLine() *clpb.CommandLine {
 }
 
 func configureGlobalCredentialHelper(ctx context.Context) error {
-	if !strings.HasPrefix(*targetRepoURL, "https://") {
+	baseURL := *pushedRepoURL
+	isFork := *targetRepoURL != "" && *pushedRepoURL != *targetRepoURL
+	if isFork {
+		baseURL = *targetRepoURL
+	}
```

Comment:
nit: this logic is duplicated in a few places, might be worth extracting a function